kv/kvserver: TestLeasePreferencesDuringOutage failed

kv/kvserver.TestLeasePreferencesDuringOutage [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3432858&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3432858&tab=artifacts#/) on master @ [079adde2ecb6f5f188cbe29346b400b49b8f550b](https://github.com/cockroachdb/cockroach/commits/079adde2ecb6f5f188cbe29346b400b49b8f550b):


```
=== RUN   TestLeasePreferencesDuringOutage
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLeasePreferencesDuringOutage900732031
    test_log_scope.go:74: use -show-logs to present logs inline
    client_lease_test.go:653: 
        	Error Trace:	client_lease_test.go:653
        	            				client_lease_test.go:659
        	Error:      	Received unexpected error:
        	            	heartbeat failed on epoch increment
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.init
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/liveness.go:59
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6309
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6286
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6286
        	            	  | runtime.main
        	            	  | 	/usr/local/go/src/runtime/proc.go:208
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1371
        	            	Wraps: (2) heartbeat failed on epoch increment
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestLeasePreferencesDuringOutage
    panic.go:632: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLeasePreferencesDuringOutage900732031
--- FAIL: TestLeasePreferencesDuringOutage (9.22s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestLeasePreferencesDuringOutage PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/kv lunevalex
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeasePreferencesDuringOutage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestLeasePreferencesDuringOutage [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3821961&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3821961&tab=artifacts#/) on master @ [40e453878a716be01d7e0fefd615b3b64cc06cd7](https://github.com/cockroachdb/cockroach/commits/40e453878a716be01d7e0fefd615b3b64cc06cd7):


```
=== RUN   TestLeasePreferencesDuringOutage
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLeasePreferencesDuringOutage821452511
    test_log_scope.go:80: use -show-logs to present logs inline
    client_lease_test.go:653: 
        	Error Trace:	client_lease_test.go:653
        	            				client_lease_test.go:659
        	Error:      	Received unexpected error:
        	            	heartbeat failed on epoch increment
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.init
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/liveness.go:60
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6498
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6475
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6475
        	            	  | runtime.main
        	            	  | 	/usr/local/go/src/runtime/proc.go:238
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (2) heartbeat failed on epoch increment
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestLeasePreferencesDuringOutage
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLeasePreferencesDuringOutage821452511
--- FAIL: TestLeasePreferencesDuringOutage (9.35s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #72920 kv/kvserver: TestLeasePreferencesDuringOutage failed [C-test-failure O-robot branch-release-21.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeasePreferencesDuringOutage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestLeasePreferencesDuringOutage [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3894809&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3894809&tab=artifacts#/) on master @ [22df0a6658a927e7b65dafbdfc9d790500791993](https://github.com/cockroachdb/cockroach/commits/22df0a6658a927e7b65dafbdfc9d790500791993):


```
=== RUN   TestLeasePreferencesDuringOutage
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLeasePreferencesDuringOutage1971605531
    test_log_scope.go:80: use -show-logs to present logs inline
    client_lease_test.go:653: 
        	Error Trace:	client_lease_test.go:653
        	            				client_lease_test.go:659
        	Error:      	Received unexpected error:
        	            	heartbeat failed on epoch increment
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.init
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/liveness.go:60
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6498
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6475
        	            	  | runtime.doInit
        	            	  | 	/usr/local/go/src/runtime/proc.go:6475
        	            	  | runtime.main
        	            	  | 	/usr/local/go/src/runtime/proc.go:238
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (2) heartbeat failed on epoch increment
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
        	Test:       	TestLeasePreferencesDuringOutage
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLeasePreferencesDuringOutage1971605531
--- FAIL: TestLeasePreferencesDuringOutage (9.01s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #72920 kv/kvserver: TestLeasePreferencesDuringOutage failed [C-test-failure O-robot branch-release-21.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeasePreferencesDuringOutage.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_lease_test.go
+++ b/pkg/kv/kvserver/client_lease_test.go
@@ -577,6 +577,8 @@ func TestLeasePreferencesDuringOutage(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	skip.UnderStress(t, "https://github.com/cockroachdb/cockroach/issues/70113")
+
 	stickyRegistry := server.NewStickyInMemEnginesRegistry()
 	defer stickyRegistry.CloseAllStickyInMemEngines()
 	ctx := context.Background()

